src/fmt.jl

```diff
@@ -79,6 +79,9 @@ default_spec(::Type{<:AbstractString}) = DEFAULT_FORMATTERS[AbstractString]
 default_spec(::Type{<:AbstractChar})   = DEFAULT_FORMATTERS[AbstractChar]
 default_spec(::Type{<:AbstractIrrational}) = DEFAULT_FORMATTERS[AbstractIrrational]
 default_spec(::Type{<:Number})         = DEFAULT_FORMATTERS[Number]
+default_spec(::Complex{T} where T<:Integer)       = DEFAULT_FORMATTERS[Complex{T} where T<:Integer]
```

Comment:
- I had tried exactly that before, but I received
```
julia> hh(x::Type{Complex{<:Integer}}) = show(typeof(x))

hh (generic function with 1 method)

julia> hh(2 + 3im)
ERROR: MethodError: no method matching hh(::Complex{Int64})
Closest candidates are:
  hh(::Type{Complex{#s12} where #s12<:Integer}) at REPL[114]:1
Stacktrace:
 [1] top-level scope at REPL[115]:1

julia> gg(x::Complex{<:Integer}) = show(typeof(x))
gg (generic function with 1 method)

julia> gg(2 + 3im)
Complex{Int64}
```
So I accepted that `Type{Complex{<:Integer}}` is a type of a type, which is what we don't want ...
I am happy to accept any prettier solution.
- In much the same way, I think, the fallback should be `<:Complex` instead of `Type{<:Complex}`
```
julia> typeof(2 + im) <: Type{<:Complex}
false

julia> typeof(2 + im) <: Complex
true
```
But I consider myself still a beginner in type programming, so please fell free to jump in, if you have a better way of putting it.
- I wrote `Complex{T} where T<:Integer` because, strangely the `Dict` does not test on identity before adding a new type:
```
julia> d = Dict{Type{T} where T, String}()
Dict{Type,String} with 0 entries

julia> d[Complex{T}  where T <: Integer] = "test"
"test"

julia> d[Complex{S}  where S <: Integer] = "test"
"test"

julia> d[Complex{<:Integer}] = "test"
"test"

julia> d
Dict{Type,String} with 3 entries:
  Complex{#s15} where #s15<:Integer => "test"
  Complex{T} where T<:Integer       => "test"
  Complex{S} where S<:Integer       => "test"

julia> (Complex{T} where T<:Integer) == (Complex{S} where S<:Integer)
true
```
So `default_fmt!(Complex{<:Integer}, 's')` will just add another entry to the dict instead of overwriting. That is also the reason, why - in the first run - I chose to define the type `ComplexInteger = Complex{T} where T<:Integer`.